.circleci/config.yml

```diff
@@ -85,11 +115,72 @@ jobs:
           path: frontend/reports/
       - store_artifacts:
           path: frontend/coverage/
+  deploy:
+    executor: docker-executor
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Build backend assets
+          command: yarn build
+      - run:
+          name: Build frontend assets
+          command: cd frontend && yarn build
+      - run: curl -v -L -o cf-cli_amd64.deb 'https://cli.run.pivotal.io/stable?release=debian64&source=github'
+      - run: sudo dpkg -i cf-cli_amd64.deb
+      - when:  # sandbox - this will be removed before PR#23 is merged
+          condition:
+            and:
+              - equal: [<< pipeline.project.git_url >>, << pipeline.parameters.dev_git_url >>]
+              - equal: [<< pipeline.git.branch >>, << pipeline.parameters.sandbox_git_branch >>]
+          steps:
+            - run:
+                name: Login with service account
+                command: cf login -a << pipeline.parameters.cg_api >> -u $CLOUDGOV_SANDBOX_USERNAME -p $CLOUDGOV_SANDBOX_PASSWORD -o << pipeline.parameters.cg_org >> -s ohstta-sandbox
+            - run:
+                name: Push application with sandbox vars
+                command: cf push --vars-file deployment_config/dev_vars.yml
```

Comment:
Is there any way to pull these steps out into something common to be used between all the environments with only parameters changing some way?

Reply:
I don't think so. I can't use a common username/password because staging and prod are going to share a project space and therefore share a set of env variables. So, the thing I need to change is really the variable names of the username and password ( `$CLOUDGOV_STAGING_USERNAME` etc.). YAML doesn't exactly lend itself to functional programming, but there is a way to create a command. (Like what I did for `create_combined_yarnlock`) but I'm assuming this won't work to construct variables because `"$CLOUDGOV_PROD_USERNAME"` and `$CLOUDGOV_PROD_USERNAME` shouldn't be the same type.  Maybe they are?  I can try it. It seems like it's just going to make things harder to debug and read though. Is there another way to do this that I'm missing?

Reply:
Are you saying https://circleci.com/docs/2.0/reusing-config/#using-the-parameters-declaration wouldn't work here?

Reply:
I think we might be talking past each other and we should brainstorm this sync. I'll follow up with you outside of gh.

Reply:
For posterity... Josh had a great idea to move the login and push steps to a command!  Apparently you _can_ store environment variables in parameters and have the environment variables stay secret. We tried echoing to stdout and circle ci still knew to hide the parameter/environment var value.